### PR TITLE
fix: handle out-of-bounds page parameter gracefully (#1103)

### LINKED ISSUE #1103: Incorrect meta when page query param is out of bound
When passing a page in a query param that is out of bound (eg page 200 for 27 total database entries), the properties `totalItems` `currentPage` `totalPages` are incorrect:

- Result with page=2 as query param
Image

- Result with page=200 as query param
Image

However, the properties are correct when using `paginationType: PaginationType.LIMIT_AND_OFFSET` AND defining relations on the configuration (still incorrect without relations)

### PR DESCRIPTION
**Pull request by tejasdev**
## Description
Fixes out-of-bounds page parameter handling by clamping currentPage to valid range.

## Changes
- Clamp `currentPage` to range [1, totalPages]
- Update links generation to use clamped `currentPage`
- Add test coverage for edge cases

## Related Issue
Closes #1103

## Testing
- Added new test cases for page > totalPages and page <= 0
- All existing tests pass

### CODE REVIEW

File: src/paginate.spec.ts [hunk 1]
```diff
@@ -3135,6 +3135,37 @@ describe('paginate', () => {
...
+        }
+        const query: PaginateQuery = {
+            path: '/toys',
+            page: 200,
+        }
+
+        const result = await paginate<ToyShopEntity>(query, toyShopRepo, config)
+
+        expect(result.links.current).toBe('/toys?page=1&limit=10&sortBy=id:ASC')
+    })
```
[the remark this answers is not in the document]
Comment (tejasdev) [reply src/paginate.spec.ts]: Not necessary as the page will be either `query.page` or `1`
